### ISSUE TITLE
Set response status message.

### ISSUE THREAD

**Issue by sholladay** [hapijs/hapi#3237]
In creating an [h2o2](https://github.com/hapijs/h2o2) proxy, I discovered that while status codes are respected, the status message is not. This matters for proxies that wish to be very transparent, particularly for servers that wish to be more user friendly and go above and beyond the generic defaults for each HTTP response code.

Reporting this here as hapi doesn't seem to have a way to set the status message.

For example:

```js
function mapResponseData(from, to) {
    // This copies the status code successfully.
    to.code(from.statusCode);
    // This is an example of how one might hypothetically copy
    // the status message. Doesn't currently do anything.
    to.statusMessage = from.statusMessage;
} 
function onResponse(err, inResponse, inRequest, reply) {
    const outResponse = reply('foo');
    mapResponseData(inResponse, outResponse);
}
```

**Comment by hueniverse** [comment]
I am not a fan of the idea of using the HTTP status message for relaying information. That said, I don't have a strong objection to providing an api such as `response.message()` to set it.

**Comment by hueniverse** [comment]
Closed by 0b1bf802d1ee1cdc554f884eb6ccc006d7b92c1e

**Comment by sholladay** [comment]
Thank you. This helps a lot, as it is important that my proxy be transparent.
